### PR TITLE
Have `$selected()` and `$root()` return tags. Remove `$asTags()` and `$get()`

### PR DESCRIPTION
**Pull request by schloerke**
Fixes https://github.com/rstudio/htmltools/issues/221
Fixes https://github.com/rstudio/htmltools/issues/213

### CODE REVIEW

**Comment by wch** [comment]
After this change, is there a way to get the root or selected tag-env objects?

**Comment by cpsievert** [comment]
> After this change, is there a way to get the root or selected tag-env objects?

If users really need this ability (do they?), perhaps we could have `$_root()` and `$_selected()` methods (or maybe add a `asTagEnv = FALSE` argument to each)?

**Comment by schloerke** [comment]
They can access them using:

* Selected elements: `$each(func)` 
* Root elements: `$reset()$each(func)`

File: R/tag_query.R [hunk 1]
```diff
@@ -377,14 +377,14 @@ print.htmltools.tag.query <- function(x, ...) {
 format.htmltools.tag.query <- function(x, ...) {
   stop(
     "`format.htmltools.tag.query(x)` not allowed.\n",
-    "Please call `format()` the result of `$asTags()`"
+    "Please call `format()` the result of `$selected()`"
```
Comment (cpsievert) [inline R/tag_query.R:380]: "`tagQuery()` objects can not be written directly as HTML tags.\n", 
    "Call either `$root()` or `$selected()` to extract the tags of interest."

File: R/tag_query.R [hunk 2]
```diff
@@ -377,14 +377,14 @@ print.htmltools.tag.query <- function(x, ...) {
...
   )
 }
 #' @export
 as.character.htmltools.tag.query <- function(x, ...) {
   stop(
     "`as.character.htmltools.tag.query(x)` not allowed.\n",
-    "Please call `as.character()` the result of `$asTags()`"
+    "Please call `as.character()` the result of `$selected()`"
```
Comment (cpsievert) [inline R/tag_query.R:387]: "`tagQuery()` objects can not be written directly as HTML tags.\n", 
    "Call either `$root()` or `$selected()` to extract the tags of interest."

**Review by cpsievert — APPROVED** [review]
Thanks!

**Comment by wch** [comment]
I don't think it would be commonly used to access them directly. I could see accessing tag-envs as being useful for debugging, and possibly for doing operations that don't yet have `tagQuery` convenience functions, but I don't know for sure. The `each(func)` method seems sufficient for that, I think.

**Pull request merged**